### DISCUSSION TITLE
SVG bounding box error when rendering Jupyter notebook with Julia code blocks

### DISCUSSION THREAD

**Discussion by rhwrhw** [quarto-dev/quarto-cli#3283] [Q&A] [answered]
I am using Quarto 1.2.207 and Jupyter Notebook (6.5.2), Python 3.9.6 (default, Aug  5 2022, 15:21:02) 
[Clang 14.0.0 (clang-1400.0.29.102)].

I am using the elsevier template, invoked as "quarto use template quarto-journals/elsevier"

The kernel is Julia 1.8.2.

There is a code cell that creates and displays a VegaLite graphic, as follows:

#| label: fig-boxplots
#| fig-cap: "Boxplots comparing EVI2 to NBR distributions at each age."

u = tidybox |> @vlplot(:boxplot, y=:value, x=:year, column=:index)
display(u)

where tidybox is a dataframe read from a csv file.

when I render I get the error below. I have tried nearly everything, including changing pdf-engines, ensuring I have Inkscape in the path, etc.

"LaTeX Error: Cannot determine size of graphic in hpf_files/figure-pdf/fig-boxplots-output-1.svg (no BoundingBox).

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.370 ...les/figure-pdf/fig-boxplots-output-1.svg}"
                                                   
The same thing also occurs if I render the qmd file.

Relevant YAML header options are as follows:

header-includes:
  - '\usepackage[left]{lineno}'
  - \linenumbers
pdf-engine: xelatex
pdf-engine-opt: '--shell-escape'
jupyter: julia-1.8

Any help would be appreciated.

**Comment by cscheid** [comment]
We should first start by making sure you're using the latest v1.2 prerelease, which is [v1.2.269](https://github.com/quarto-dev/quarto-cli/releases/tag/v1.2.269). But I think the issue is ultimately coming the fact that the library you're using (whatever has the `@vlplot` macro) doesn't seem to be emitting width and height values with the SVG. I believe those are necessary for the PDF engine to determine the size of the figure.

I haven't used this library and I will need a fully reproducible example to test any of this on my side, but you could try to have your plotting library emit a high-resolution PNG, which isn't an ideal solution but would probably work.

**Reply by rhwrhw** [reply]
Hi, thanks for picking this up. I did just upgrade to v1.1.269, but that did not help. I failed to mention that this is on a Mac with the M1 Max and macOS Monterey v. 12.6. I just shared a repository with the fully reproducible example you requested.

**Reply by compleathorseplayer** [reply]
I have the same issue (older Mac).  [I was going to recommend quarto to my students for use in class until I came across this (and will if I can find a resolution)].  Thanks for any insights.

**Comment by kjohnsen** [comment]
I'm actually having an issue including just any SVG in LaTeX output, even with height and width defined. I'm using Quarto 1.2.280 and just barely upgraded TinyTex to v2022.12.12.

`test.svg`:
```svg
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
  <circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow" />
</svg>
```

`test.qmd`:
```qmd
# hello there

![test SVG](test.svg)
```

Render, and for me it complains about the bounding box:
```sh
quarto render test.qmd --to pdf
```

```
LaTeX Error: Cannot determine size of graphic in test.svg (no BoundingBox).

See the LaTeX manual or LaTeX Companion for explanation.
```

**Reply by compleathorseplayer** [reply]
This is the only thing keeping me from using Quarto in my (200+ size) (university postgrad) classes - if this were fixed, I would require every student to install and use Quarto.

**Reply by kjohnsen** [reply]
@cscheid Maybe my example belongs more in issues than discussions. I'll post it there

**Reply by cderv** [reply]
See answer below https://github.com/quarto-dev/quarto-cli/discussions/3283#discussioncomment-5507523

**Comment by hillyuan** [comment]
Same problem (LaTex no BoundingBox error) still exists in version 1.3.302 with tinytex v2023.04 in Windows 11.

**Reply by cderv** [reply]
See answer below https://github.com/quarto-dev/quarto-cli/discussions/3283#discussioncomment-5507523

**Reply by hillyuan** [reply]
Thanks for your help. All my svg files could be transferred to PDF successfully following your instruction. However, when including gif animation file, I still git similar error like: 

`LaTeX Error: Cannot determine size of graphic in ./img/wave1D.gif (no BoundingBox)`

**Reply by cderv** [reply]
Yes GIF are different than `.svg`. They can't be included easily in LaTeX. See discussion about potential solutions in https://github.com/quarto-dev/quarto-cli/discussions/3551

There is no automatic conversion for that in Quarto.

**Reply by hillyuan** [reply]
I see! Thanks your for the notice.

**Comment by cderv** [comment, marked as answer]
Quarto should now handle SVG image in PDF output format, by converting the SVG file to PDF file for inclusion. This is a new feature of Quarto 1.3 and **it requires `rsvg-convert` CLI tool**. 

This is documented for now a 1.3 release feature : https://quarto.org/docs/prerelease/1.3/pdf.html#svg-images  with some indication on how to install `rsvg-convert` for each OS. 

Taking the example from @kjohnsen at https://github.com/quarto-dev/quarto-cli/discussions/3283#discussioncomment-4551997

````markdown
---
title: Testing SVG
format: pdf
---

# hello there

![test SVG](test.svg)

````

with `test.svg` being 
````svg
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
  <circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow" />
</svg>
````

This should render without error now, the `test.svg` file will be converted on the fly to `test.pdf` to be included using `\includegraphics`

![image](https://user-images.githubusercontent.com/6791940/229462153-b75dfdef-67b6-44c1-9d0e-55cd8627e168.png)


For context, this was done in https://github.com/quarto-dev/quarto-cli/issues/2575
